core/src/main/java/com/graphhopper/util/PathSimplification.java

```diff
@@ -141,7 +141,12 @@ public PointList simplify() {
     private int getLength(Object o, int index) {
         if (o instanceof InstructionList) {
             // we do not store the last point of an instruction
-            int size = ((InstructionList) o).get(index).getPoints().size();
+            Instruction instruction = ((InstructionList) o).get(index);
+            int size;
+            if(instruction instanceof ViaInstruction)
+                size = 0;
+            else
+                size = instruction.getPoints().size();
```

Comment:
Here I would expect 'getPoints.size-1'

Reply:
For the ViaInstruction it shouldn't matter as it should be only one point anyway. For the other case it should be the size IMHO. Could you explain why it should be size-1?

Reply:
Consistency. All path details currently work this way: you specify first and last and the length is last-first. Now imagine an instruction 0-1-2-3-4, where e.g. at 3 there is the via point, then the first instruction includes point 3, i.e. it is `interval:[0,3]` and therefor the length is 3-0=3, although the pointlist.size is 4.